### PR TITLE
Add more match test cases on paths.

### PR DESCRIPTION
**Pull request by xtcyclist**
## What type of PR is this?
- [ ] bug
- [ ] feature
- [X] enhancement

## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

**Review by yixinglu — APPROVED** [review]

File: tests/tck/features/match/Path.feature [hunk 1]
```diff
@@ -38,10 +38,296 @@ Feature: Matching paths
...
+      | count(p) |
+      | 28       |
+    When executing query:
+      """
+      match p2 = (v2:Label_5)-[e2]-(v2)
+      return count(p2)
+      """
+    Then the result should be, in any order:
+      | count(p2) |
+      | 43        |
```
Comment (czpmango) [inline tests/tck/features/match/Path.feature:57]: Any difference between these two test cases?

File: tests/tck/features/match/Path.feature [hunk 2]
```diff
@@ -72,3 +359,132 @@ Feature: Matching paths
...
+      | count(*) |
+      | 16       |
+    When executing query:
+      """
+      match (v1)-[e*1..2]->(v2)
+      where id(v1) in [6]
+      match (v2)-[e*1..1]->(v1)
+      return count(*)
+      """
+    Then a SemanticError should be raised at runtime: e binding to different type: Edge vs EdgeList
```
Comment (czpmango) [inline tests/tck/features/match/Path.feature:490]: Please retest this case.

**Review by jievince — APPROVED** [review]

**Comment by czpmango** [comment]
Please test the relationship isomorphism for multiple path matching.

File: tests/tck/features/match/Path.feature (hunk 1 above)
Comment (xtcyclist) [reply tests/tck/features/match/Path.feature:57]: These count results are used to prove the correctness of the following query.

File: tests/tck/features/match/Path.feature (hunk 2 above)
Comment (xtcyclist) [reply tests/tck/features/match/Path.feature:490]: Still reporting this error.

Comment (czpmango) [reply tests/tck/features/match/Path.feature:490]: This is really a bug...

**Pull request merged**